sdk/core/azure-core-http-httpurlconnection/src/main/java/com/azure/core/http/httpurlconnection/HttpUrlConnectionAsyncClient.java

```diff
@@ -0,0 +1,330 @@
+import reactor.core.publisher.Mono;
+import com.azure.core.http.HttpHeader;
+import reactor.core.scheduler.Schedulers;
+
+import java.io.*;
+import java.net.*;
+import java.nio.ByteBuffer;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.time.Duration;
+import java.util.*;
+
+/**
+ * This class provides a HttpUrlConnection implementation for the {@link HttpClient} interface.
+ * Creating an instance of this class can be achieved by using the {@link HttpUrlConnectionAsyncClientBuilder} class.
+ *
+ * @see HttpClient
+ * @see HttpUrlConnectionAsyncClientBuilder
+ */
+public class HttpUrlConnectionAsyncClient implements HttpClient {
+    private static final ClientLogger LOGGER = new ClientLogger(HttpUrlConnectionAsyncClient.class);
+    private final long connectionTimeout; // in milliseconds format for HttpUrlConnection methods
+    private final long readTimeout; // in milliseconds format for HttpUrlConnection methods
+    private final Duration writeTimeout;
+    private final Duration responseTimeout;
+    private final ProxyOptions proxyOptions;
+    private final Configuration configuration;
+
+    HttpUrlConnectionAsyncClient(HttpUrlConnectionTimeouts timeouts, ProxyOptions proxyOptions, Configuration configuration) {
+        this.connectionTImeout = timeouts.connectionTimeout == null ? -1 : timeouts.connectionTimeout.toMillis();
+        this.readTimeout = timeouts.readTimeout == null ? -1 : timeouts.readTimeout.toMillis();
+        this.writeTimeout = timeouts.writeTimeout;
+        this.responseTimeout = timeouts.responseTimeout;
+
+        this.proxyOptions = proxyOptions;
+        this.configuration = configuration;
+    }
+
+    @Override
+    public Mono<HttpResponse> send(HttpRequest httpRequest) {
+        return sendAsync(httpRequest, Context.NONE);
+    }
+
+    @Override
+    public Mono<HttpResponse> send(HttpRequest request, Context context) {
+        return sendAsync(request, context);
+    }
+
+    @Override
+    public HttpResponse sendSync(HttpRequest httpRequest, Context context) {
+        ProgressReporter progressReporter = Contexts.with(context).getHttpRequestProgressReporter();
+
+        if (httpRequest.getHttpMethod() == HttpMethod.PATCH) {
+            return sendPatchViaSocketSync(httpRequest);
+        }
+
+        HttpURLConnection connection = connect(httpRequest);
+        sendSyncRequest(httpRequest, progressReporter, connection);
+        return receiveResponse(httpRequest, connection);
+    }
+
+    /**
+     * Asynchronously send the HttpRequest
+     *
+     * @param httpRequest The HTTP Request being sent
+     * @param context The context of the request, for any additional changes
+     * @return A Mono containing a HttpResponse object
+     */
+    private Mono<HttpResponse> sendAsync(HttpRequest httpRequest, Context context) {
+        ProgressReporter progressReporter = Contexts.with(context).getHttpRequestProgressReporter();
```

Comment:
This is not used if the HTTP method is PATCH, so you could move this to after the following conditional statement.